### ISSUE TITLE
Porting Squeezebert model to n300

### ISSUE THREAD

**Issue by saichandax** [tenstorrent/tt-metal#13328]
- [x] Measure and record current performance.
- [x] Rebase the model to main, ensure the PCC = 0.99
- [x] Port functionality to n300 card (single device)
- [x] Provide Op Report
- [ ] Check Model into CI
- [x] Implement Demo
- [ ] #13397
- [ ] #14727

Executive summary **(as of Nov 12)**:
- Single Device implementation is complete: **(BLOCKED) due to pending approvals for [#9371](https://github.com/tenstorrent/tt-metal/pull/9371)**
    - PCC = 0.99
    - 24 torch convs ([#13195](https://github.com/tenstorrent/tt-metal/pull/13195))
    - BS = 8
    - Perf numbers
        - E2E perf is 1.27 samples/sec
        - Device perf is 185.26 samples/sec

- Data parallel implementation is complete: **(BLOCKED) due to pending approvals and CIs for [#13418](https://github.com/tenstorrent/tt-metal/pull/13418)**
    - PCC = 0.99
    - 24 torch convs ([#13195](https://github.com/tenstorrent/tt-metal/pull/13195))
    - BS = 16
    - Perf numbers
        - E2E perf is 1.529 samples/sec
        - Device perf is 380.56 samples/sec

- Trace_2cqs implementation is not done yet (BLOCKED due torch ops in the model)

ToDo:
- Update perf numbers and merge single device and data parallel implementation once we have PR approvals.
- To implement perf with trace_2cqs (currently blocked by ([#13195](https://github.com/tenstorrent/tt-metal/pull/13195)))

**Comment by saichandax** [comment]
Closing the issue. No longer relevant.
